### PR TITLE
feat: plugin config write guide

### PR DESCRIPTION
**Pull request by JimMoen**
related: https://github.com/emqx/emqx-plugin-template/pull/126

### CODE REVIEW

File: zh_CN/extensions/plugins.md [hunk 1]
```diff
@@ -163,6 +165,142 @@ make rel
 
 这将创建一个新的 EMQX 插件 tarball `my_emqx_plugin-1.0.0.tar.gz`，您现在可以上传并安装到运行中的 EMQX 集群中。
 
+#### 为插件编写 Config Schema （可选）
```
Comment (wivwiv) [inline zh_CN/extensions/plugins.md:168]:
```suggestion
#### 为插件编写 Config Schema （可选）

{% emqxce %}
:::tip
EMQX 企业版功能。EMQX 企业版可以为您带来更全面的关键业务场景覆盖、更丰富的数据集成支持，更高的生产级可靠性保证以及 24/7 的全球技术支持，欢迎[免费试用](https://www.emqx.com/zh/try?product=enterprise)。
:::
{% endemqxce %}
```

File: en_US/extensions/plugins.md [hunk 2]
```diff
@@ -162,6 +164,142 @@ make rel
 
 This created a new EMQX plugin tarball  `my_emqx_plugin-1.0.0.tar.gz` that you can now upload and install to your running EMQX cluster.
 
+#### Write Config Schema for the plugin (Optional)
```
Comment (wivwiv) [inline en_US/extensions/plugins.md]:
```suggestion
#### Write Config Schema for the plugin (Optional)

{% emqxce %}
:::tip
EMQX Enterprise Edition features. EMQX Enterprise Edition provides comprehensive coverage of key business scenarios, rich data integration, product-level reliability, and 24/7 global technical support. Experience the benefits of this [enterprise-ready MQTT messaging platform](https://www.emqx.com/en/try?product=enterprise) today.
:::
{% endemqxce %}
```

**Review by wivwiv — CHANGES_REQUESTED** [review]

File: en_US/extensions/plugins.md [hunk 3]
```diff
@@ -162,6 +164,154 @@ make rel
...
+:::tip
+EMQX Enterprise Edition features. EMQX Enterprise Edition provides comprehensive coverage of key business scenarios, rich data integration, product-level reliability, and 24/7 global technical support. Experience the benefits of this [enterprise-ready MQTT messaging platform](https://www.emqx.com/en/try?product=enterprise) today.
+:::
+{% endemqxce %}
+
+{% emqxce %}
+::: tip Tip
+EMQX Enterprise Edition features. EMQX Enterprise Edition provides comprehensive coverage of key business scenarios, rich data integration, product-level reliability, and 24/7 global technical support. Experience the benefits of this [enterprise-ready MQTT messaging platform](https://www.emqx.com/en/try?product=enterprise) today.
+:::
+{% endemqxce %}
```
Comment (zmstone) [inline en_US/extensions/plugins.md]: this part is repeated.

File: en_US/extensions/plugins.md [hunk 4]
```diff
@@ -162,6 +164,154 @@ make rel
...
+
+We introduced Avro Schema for plugins in EMQX Enterprise Edition 5.7.0 to provide the ability to update plugin configurations at runtime.
+Furthermore, you can declare UI in this schema, allowing the EMQX Dashboard to render configuration forms for ease of use.
```
Comment (zmstone) [inline en_US/extensions/plugins.md]: The initial load of the plugin config seems to be missed in the feature implementation ?

```suggestion
Starting with version 5.7, the EMQX plugin framework pre-loads JSON or HOCON format configuration file for each plugin before startup and validates them against an optional schema, if provided. The configuration schema should be described using an Avro schema, located in the file priv/plugin_name.avsc. Additionally, configurations can be updated at runtime by utilizing the management API with the request PUT /plugins/:name/config.

For the EMQX Enterprise edition, users have the option to include $ui metadata within the Avro schema objects. This metadata aids in generating a configuration management page on the dashboard, enhancing user interaction and ease of management.
```

File: en_US/extensions/plugins.md [hunk 5]
```diff
@@ -162,6 +164,154 @@ make rel
...
+
+::: tip **Tip**
+
+You can find two example files in the project directory: `priv/config_schmea.avsc.example`, `priv/config_i18n.json.example`.
```
Comment (zmstone) [inline en_US/extensions/plugins.md]:
```suggestion
You can find two example files in the [template project](https://github.com/emqx/emqx-plugin-template) directory: `priv/config_schmea.avsc.example`, `priv/config_i18n.json.example`.
```

**Review by zmstone — DISMISSED** [review]
please also add:
1. Where should the initial config (before EMQX boot) be provided (maybe in the priv dir)
2. Where are the updated config file stored, (explain overwrite, and the bak files too?)

File: en_US/extensions/plugins.md [hunk 6]
```diff
@@ -162,70 +175,64 @@ make rel
...
-{% emqxce %}
-::: tip Tip
 EMQX Enterprise Edition features. EMQX Enterprise Edition provides comprehensive coverage of key business scenarios, rich data integration, product-level reliability, and 24/7 global technical support. Experience the benefits of this [enterprise-ready MQTT messaging platform](https://www.emqx.com/en/try?product=enterprise) today.
+
 :::
 {% endemqxce %}
 
-We introduced Avro Schema for plugins in EMQX Enterprise Edition 5.7.0 to provide the ability to update plugin configurations at runtime.
-Furthermore, you can declare UI in this schema, allowing the EMQX Dashboard to render configuration forms for ease of use.
+EMQX Enterprise version 5.7.0 introduced Avro Schema for plugins, enhancing the ability to update plugin configurations dynamically during runtime. This feature also allows for the declaration of a user interface in the schema, enabling the EMQX Dashboard to dynamically generate configuration forms for easier management.
```
Comment (zmstone) [inline en_US/extensions/plugins.md]: Avro schema is available for opensource edition, the UI for configuration management is provided in Enterprise only.

```suggestion
EMQX version 5.7.0 introduced REST API for plugin configuration management and Avro Schema for configuration validation, enhancing the ability to update plugin configurations dynamically during runtime.
For Enterprise edition, it also allows the declaration of a user interface in the schema, enabling the EMQX Dashboard to dynamically generate configuration forms for easier management.
```

File: en_US/extensions/plugins.md [hunk 7]
```diff
@@ -48,7 +49,9 @@ my_emqx_plugin
 ├── erlang_ls.config
 ├── get-rebar3
 ├── priv
-│   └── config.hocon
+│   ├── config.hocon
+│   ├── config_i18n.json.example
+│   └── config_schmea.avsc.example
```
Comment (zhongwencool) [inline en_US/extensions/plugins.md]: typo: schema

File: zh_CN/extensions/plugins.md [hunk 8]
```diff
@@ -163,6 +165,145 @@ make rel
...
+
+{% emqxce %}
+:::tip
+EMQX 企业版功能。EMQX 企业版可以为您带来更全面的关键业务场景覆盖、更丰富的数据集成支持，更高的生产级可靠性保证以及 24/7 的全球技术支持，欢迎[免费试用](https://www.emqx.com/zh/try?product=enterprise)。
+:::
+{% endemqxce %}
+
+我们在 EMQX 企业版 5.7.0 中为插件引入了 Avro Schema 以提供在运行时更新插件配置的能力，并且可以在该 Schema 中提供 UI 声明，从而可以在 EMQX Dashboard 中渲染配置表单以便使用。
```
Comment (wivwiv) [inline zh_CN/extensions/plugins.md]:
```suggestion
我们在 EMQX v5.7.0 中引入了用于插件配置管理的 REST API 和用于配置验证的 Avro Schema，增强了在运行期间动态更新插件配置的能力。
对于企业版，它还允许在 Schema 中声明配置界面，从而使 EMQX Dashboard 能够动态生成配置表单，方便插件的配置和管理。
```

**Review by wivwiv — CHANGES_REQUESTED** [review]

**Review by wivwiv — DISMISSED** [review]

**Review by wivwiv — APPROVED** [review]

**Pull request merged**